crates/chia-consensus/src/check_time_locks.rs

```diff
@@ -0,0 +1,340 @@
+                return Some(ErrorCode::AssertMyBirthHeightFailed);
+            }
+        }
+        if let Some(birth_seconds) = spend.birth_seconds {
+            if birth_seconds != unspent.timestamp {
+                return Some(ErrorCode::AssertMyBirthSecondsFailed);
+            }
+        }
+        if let Some(height_relative) = spend.height_relative {
+            if prev_transaction_block_height < unspent.confirmed_block_index + height_relative {
+                return Some(ErrorCode::AssertHeightRelativeFailed);
+            }
+        }
+        if let Some(seconds_relative) = spend.seconds_relative {
+            if timestamp < unspent.timestamp + seconds_relative {
+                return Some(ErrorCode::AssertSecondsRelativeFailed);
+            }
+        }
+        if let Some(before_height_relative) = spend.before_height_relative {
+            if prev_transaction_block_height
+                >= unspent.confirmed_block_index + before_height_relative
+            {
+                return Some(ErrorCode::AssertBeforeHeightRelativeFailed);
+            }
+        }
+        if let Some(before_seconds_relative) = spend.before_seconds_relative {
+            if timestamp >= unspent.timestamp + before_seconds_relative {
+                return Some(ErrorCode::AssertBeforeSecondsRelativeFailed);
+            }
+        }
+    }
+
+    None
+}
+
+#[cfg(feature = "py-bindings")]
+#[pyfunction]
+#[pyo3(name = "check_time_locks")]
+#[allow(clippy::needless_pass_by_value)] // pyo3 prefers pass_by_value
+pub fn py_check_time_locks(
+    removal_coin_records: HashMap<chia_protocol::BytesImpl<32>, CoinRecord>,
+    bundle_conds: &OwnedSpendBundleConditions,
+    prev_transaction_block_height: u32,
+    timestamp: u64,
+) -> PyResult<Option<u32>> {
+    let res = check_time_locks(
+        &removal_coin_records,
+        bundle_conds,
+        prev_transaction_block_height,
+        timestamp,
+    );
+
+    Ok(res.map(std::convert::Into::into))
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::owned_conditions::OwnedSpendConditions;
+    use chia_protocol::Coin;
+    use rstest::rstest;
+    use std::collections::HashMap;
+
+    fn dummy_coin_record(confirmed_block_index: u32, timestamp: u64) -> CoinRecord {
+        CoinRecord {
+            coin: Coin {
+                parent_coin_info: Bytes32::default(),
+                puzzle_hash: Bytes32::default(),
+                amount: 1,
+            },
+            confirmed_block_index,
+            spent_block_index: 0,
+            coinbase: false,
+            timestamp,
+        }
+    }
+
+    fn dummy_spend_conditions(
+        coin_id: Bytes32,
+        height_relative: Option<u32>,
+        seconds_relative: Option<u64>,
+        before_height_relative: Option<u32>,
+        before_seconds_relative: Option<u64>,
+        birth_height: Option<u32>,
+        birth_seconds: Option<u64>,
+    ) -> OwnedSpendConditions {
```

Comment:
I think it would be better to implement the `Default` trait than having this function.